ext/opcache/zend_accelerator_module.c

```diff
@@ -108,6 +108,19 @@ static ZEND_INI_MH(OnUpdateMaxWastedPercentage)
 	return SUCCESS;
 }
 
+static ZEND_INI_MH(OnUpdateConsistencyChecks)
+{
+	zend_long *p = (zend_long *) ZEND_INI_GET_ADDR();
+	zend_long consistency_checks = atoi(ZSTR_VAL(new_value));
+
+	if (consistency_checks != 0) {
+		zend_accel_error(ACCEL_LOG_WARNING, "opcache.consistency_checks is reset back to 0 because it does not work properly (see GH-8065, GH-10624).\n");
+		return FAILURE;
+	}
+	*p = 0;
```

Comment:
Thank you for this PR! There are two things that I'm unsure about:

* Does the `on_modify` function get called initially with the default value even if it is set in some ini file?
* Does the ini value buffer get zeroed before use?

Because these are not obvious (even if one of them is true) it might be better to set `*p` unconditionally before the error. With both of the points above being false, using a different value would not set `*p` and thus depend on uninitialized memory. This is probably not the case, but setting it might make it obvious that this can't happen.

Reply:
It gets called initially. The first line of the test is a warning. The second line is the var_dump. So this seems to me that it get called initially.

I think the ini value gets its default, because otherwise the MH's of OnUpdateMaxAcceleratedFiles etc would be wrong too?

Reply:
I checked it to be sure, and `zend_register_ini_entries_ex` does indeed always set the default value. And since `OnUpdateConsistencyChecks` follows the same structure as the other handlers (I missed that since I looked at yours in isolation) this looks fine!